spec/features/holds_spec.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Holds', type: :feature do
+  let(:mock_user) { 'patron1' }
+
+  before do
+    login_as(username: 'PATRON1', patron_key: mock_user)
+  end
+
+  after do
+    Redis.new.del('item_type_map')
+  end
+
+  it 'lets the user change the pickup library of a hold', js: true do
+    visit holds_path
+    login_as(username: 'PATRON1', patron_key: mock_user)
```

Comment:
interesting that it requires another login_as, is it like per request or sth?

Reply:
yup. It does. the login_as will work for *the next page being viewed (like http request)* which i learned the hard way through very cryptic error messages. The line I linked to from the warden repo is what clued me in. https://github.com/wardencommunity/warden/blob/5b3cbd5bef67cbe399bb7007537bc4841bbee772/lib/warden/test/helpers.rb#L14